lib/bolt/task.rb

```diff
@@ -60,15 +65,16 @@ def select_implementation(target, additional_features = [])
       mfiles = impl.fetch('files', []) + metadata.fetch('files', [])
       dirnames, filenames = mfiles.partition { |file| file.end_with?('/') }
       impl['files'] = filenames.map do |file|
-        path = file_map[file]
+        path = file_path(file)
         raise "No file found for reference #{file}" if path.nil?
         { 'name' => file, 'path' => path }
       end
 
       unless dirnames.empty?
         files.each do |file|
-          if dirnames.any? { |dirname| file['name'].start_with?(dirname) }
-            impl['files'] << file
+          name = file['name']
+          if dirnames.any? { |dirname| name.start_with?(dirname) }
+            impl['files'] << { 'name' => name, 'path' => file_path(name) }
```

Comment:
I'm confused why this is any different. Do we modify `files` somewhere?

Reply:
Ah, we do in the PuppetServer child class.

Reply:
This change feels slightly convoluted. Maybe add some comment to `file_path` explaining why it's necessary?

Reply:
can do